File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/layout/UXFlowPane.java
package org.develnext.jphp.ext.javafx.classes.layout;

import javafx.geometry.HPos;
import javafx.geometry.Orientation;
import javafx.geometry.Pos;
import javafx.geometry.VPos;
import javafx.scene.layout.FlowPane;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "layout\\UXFlowPane")
public class UXFlowPane<T extends FlowPane> extends UXPane<FlowPane> {
    interface WrappedInterface {
        @Property VPos rowValignment();
        @Property HPos columnHalignment();

        @Property double prefWrapLength();

        @Property double hgap();
        @Property double vgap();

        @Property Pos alignment();

        @Property Orientation orientation();
    }

    public UXFlowPane(Environment env, T wrappedObject) {
        super(env, wrappedObject);
    }

    public UXFlowPane(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new FlowPane();
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/layout/UXFlowPane.php
<?php
namespace php\gui\layout;

/**
 * Class UXFlowPane
 * @package php\gui\layout
 * @packages gui, javafx
 */
class UXFlowPane extends UXPane
{
    /**
     * @var string HORIZONTAL or VERTICAL
     */
    public $orientation = 'HORIZONTAL';

    /**
     * @var float
     */
    public $hgap = 0.0;

    /**
     * @var float
     */
    public $vgap = 0.0;

    /**
     * @var float
     */
    public $prefWrapLength = 400;

    /**
     * @var string
     */
    public $rowValignment = 'CENTER';

    /**
     * @var string
     */
    public $columnHalignment = 'LEFT';

    /**
     * @var string
     */
    public $alignment = 'CENTER_LEFT';
}